### PR TITLE
Add test that pre-baked datasets produce samples without padding

### PR DESCRIPTION
**Pull request by kylesayrs**
SUMMARY:
Add a unit test that runs `get_processed_dataset` over all pre-baked datasets (the dataset names registered in `src/llmcompressor/transformers/data`) and asserts the generated samples contain no padding.

The processor is obtained via `AutoProcessor` rather than `AutoTokenizer`, which lets the test also cover the multimodal pre-baked datasets by exercising only their text tokenization:
- `flickr` uses a `Qwen/Qwen3-VL-8B-Instruct` processor stub.
- `peoples_speech` uses an `openai/whisper-large-v3` processor stub.

All other pre-baked datasets (`perfectblend`, `open_platypus`, `evolcodealpaca`, `gsm8k`, `ultrachat_200k`, `wikitext`) use the tiny llama processor.

The no-padding check is feature-aware:
- Datasets emitting `input_ids` + `attention_mask` (text and `flickr`) assert every sample's attention mask is all ones.
- `peoples_speech` (whisper) emits `decoder_input_ids` and no attention mask, so it asserts the single, un-padded transcription is non-empty and contains at most one pad terminator.

TEST PLAN:
- `pytest tests/llmcompressor/transformers/data/test_dataset_loading.py::test_prebaked_datasets_generate_no_padding` — all 8 parametrized cases pass.
- `ruff check` and `ruff format` on the modified file pass.

### CODE REVIEW

File: tests/llmcompressor/transformers/data/test_dataset_loading.py [hunk 1]
```diff
@@ -344,3 +345,93 @@ def preprocess(sample):
...
+    tokenizer defines no pad token, the framework substitutes the EOS token as
+    the pad token (pad_token_id == eos_token_id), and EOS legitimately appears in
+    the text -- so that check would produce false positives.
+
+    "custom" is not a pre-baked HF dataset (it requires a local path), so it is
+    not covered here.
+    """
+    processor = AutoProcessor.from_pretrained(
+        processor_id if processor_id is not None else tiny_llama_path
+    )
```
[the remark this answers is not in the document]
Comment (kylesayrs) [reply tests/llmcompressor/transformers/data/test_dataset_loading.py:396]: That's just not true. AutoProcessor still works, even for text-only paths

File: tests/llmcompressor/transformers/data/test_dataset_loading.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import pytest
 from datasets import Dataset, IterableDataset, load_dataset
+from transformers import AutoProcessor
```
[the remark this answers is not in the document]
Comment (kylesayrs) [reply tests/llmcompressor/transformers/data/test_dataset_loading.py:3]: False

**Review by HDCharles — APPROVED** [review]
